### DISCUSSION TITLE
Week 3

### DISCUSSION THREAD

**Discussion by jakegros** [USU-ITLS-x265-23-summer/class-repo#89] [Week 3 - Discussions]
I found creating the Cafe Menu immensely helpful.  Seeing the fruits of labor really motivated me.  Compared to some JS activities, having the visual of a whole project, vs just entering a string or two, makes a huge difference for me learning and remembering code. The Javascript instructional approach needs more context.  I have a cheatsheet that has more explanation of what things are and I would include more of that with the JavaScript approach.  FreeCodeCamp's HTML and CSS approach works for me.

For learning layout with Flexbox, I loved the game-style, I'm a huge fan of gamification. This had a visual to see what my output was. More importantly, it gives context for what I am trying to do, ie. I can connect the purpose of code, variables, layout, etc that I am writing.  I've now realized why FreeCodeCamp was so useful even before this week-- it showed me what I was doing and why.
Being able to click on the properties to remember the values while working with Flexbox Froggy is brilliant.  Flexbox Froggy was directionally accurate, whereas the CSS diner was relative.  However, the live feedback for both was indispensable. 

Just a tip, GitHub and freeCodeCamp both have mobile apps.  It's not ideal and only some things are supported. However, it is still nice to have that option of connecting. Flexbox Froggy also supports mobile. 

Here's a site that helped give me some more resources and background for CSS:
https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox#horizontal_and_vertical_alignment

**Comment by metasean** [comment]
[MDN](https://developer.mozilla.org/en-US/) is the primary reference for JavaScript, and their guides are really useful for some people. 

> The Javascript instructional approach needs more context.

I completely agree. I suspect one of the reasons they didn't is that there is a lot of material to work through, and using it in projects would make it even more time to work through than it already is.

**Reply by jakegros** [reply]
Good point.  Also, as I have done, there are many resources to draw on for those who need it while completing the activities.

**Comment by derekscholes** [comment]
Thanks for the link! I never have heard of flexbox. I will definitely take a deeper look into this so that I can apply it to my project.
